functions/api_flows/app.py

```diff
@@ -767,8 +767,22 @@ def post_flow_storage_by_id(
     flow_storage_post: Flowstoragepost,
     flow_id: Annotated[str, Path(alias="flowId", pattern=UUID_PATTERN)],
 ):
-    if flow_storage_post.limit is None:
+    if flow_storage_post.limit and flow_storage_post.object_ids:
+        raise BadRequestError(
+            "Bad request. Invalid flow storage request JSON or the flow 'container' is not set. If object_ids supplied, some or all already exist."
+        )  # 400
+    if flow_storage_post.limit is None and flow_storage_post.object_ids is None:
         flow_storage_post.limit = constants.DEFAULT_PUT_LIMIT
+    # Check if any object_ids already exist in the storage table
+    if flow_storage_post.object_ids:
+        for object_id in flow_storage_post.object_ids:
+            get_item = storage_table.get_item(
+                Key={"object_id": object_id, "flow_id": flow_id}
+            )
```

Comment:
Might it be a good idea here to set a ProjectionExpression to minimise the response size from DynamoDB? (depends how large the item usually is, i think you need to return at least one key value)

Or to use batch_get_item? (code will probably be more complex, so depends how often it's likely to be iterated)